code/projects/app/src/app/shared/file-download/file-download.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { Observable } from "rxjs";
+
+type DownloadData = {
+  fileName: string;
+  path: string;
+} | null;
+
+@Component({
+  selector: "app-file-download",
+  standalone: true,
+  imports: [],
+  templateUrl: "./file-download.component.html",
+  styleUrl: "./file-download.component.css",
+})
+export class FileDownloadComponent implements OnInit {
+  @Input({ required: true }) public file$!: Observable<File>;
+
+  protected data: DownloadData = null;
+
+  public ngOnInit(): void {
+    this.file$.subscribe((file) => {
+      this.data = {
```

Comment:
If data is only a variable dependent on an observable, you may use an `async pipe` ([random example](https://v2.angular.io/docs/ts/latest/api/common/index/AsyncPipe-pipe.html), you can also search in the repo).

Reply:
I still need to `URL.createObjectURL(file)` to get downloadable file. So it's not just `file$ | async`, but I got it to work.

Reply:
Nevermind. We have to make an `Subscription` and save some variable for template to use, or we can make a `Pipe` that transforms the file to custom object.